bfh/__init__.py

```diff
@@ -79,8 +79,8 @@ def serialize(self, implicit_nulls=True):
         """
         outd = {}
         for name in self._field_names:
-            value = self.__dict__.get(name)
             field = self._fields.get(name)
+            value = getattr(self, name)
```

Comment:
we need to access the `__get__` method on the Field, not the dict, because on init, if we search in the `__dict__`, the value is not there, as it's set to `None`.